app/views/articles/show.html.erb

```diff
@@ -92,7 +92,7 @@
 
   <main class="crayons-layout__content grid gap-4">
     <div class="article-wrapper">
-      <% if !@article.published %>
+      <% if !@article.published && user_signed_in? && @article.user_id == current_user.id %>
```

Comment:
I think it might be more ideal to have `user_signed_in? && @article.user_id == current_user.id` in an `if` statement just surrounding the edit button below. I think having the warning "This URL is public but secret, so share at your own discretion." is helpful and a good reminder to everyone. 

```
if user_signed_in? && @article.user_id == current_user.id
  <a href="<%= @article.path %>/edit" class="fw-bold">Click to edit</a>.
end
```

Reply:
I agree with your suggestion😄  I fixed it in 38a9014eed62127cd2595e0007d3e41ac58d97fc.